vllm/worker/model_runner.py

```diff
@@ -1379,6 +1385,16 @@ def execute_model(
             logits=logits,
             sampling_metadata=model_input.sampling_metadata,
         )
+        if (self.observability_config.collect_model_forward_time
+                and output is not None):
+            model_forward_end.synchronize()
```

Comment:
does it do cpu <> gpu sync?

Reply:
It is a possiblity if I understand correctly. In practice though, once the sampling has finished, there should be no work pending on the gpu.